### DISCUSSION TITLE
CLOG 2.1 - Dev Journal

### DISCUSSION THREAD

**Discussion by rabbibotton** [rabbibotton/clog#337] [General]
Goals -
1) More complete coding tools. debugger, etc
2) Better auto update
3) One click install improvements
4) Faster gui builder

**Comment by rabbibotton** [comment]
One click for windows repo added https://github.com/rabbibotton/clog-win64-ez
Auto update menu option
Full interactive debugging in builder

**Comment by rabbibotton** [comment]
with-clog-debugger - any errors that with in will use a graphical debugger in clog-gui's

**Comment by rabbibotton** [comment]
Clicking sys-browser file name that is now a button opens the file in source-editor and finds the location in the file.

**Comment by rabbibotton** [comment]
clog-gui-initialize can now be set with use-clog-debugger so that all events use the clog-debugger

**Comment by rabbibotton** [comment]
Application launched from builder will by default pop debug messages on the builder page that launched the app

**Comment by rabbibotton** [comment]
There is now a console window and support for input to stdin as dialogs, so possible to use repl for console based apps etc

**Comment by rabbibotton** [comment]
clog-builder-repl added to give access to a graphical window with in the builder
clog-gui, clog-web added to clog-user

**Comment by rabbibotton** [comment]
The standard dialogs now can be set to block and return values using :time-out

**Comment by rabbibotton** [comment]
The server-file-dialog is resizable now

**Comment by rabbibotton** [comment]
The source editor now uses clog-popup what that means is that tabs or popups (configurable to use) are now slaves of your builder, so now existing windows of source will be reused and be focus on even if in different windows in the browser.

**Comment by rabbibotton** [comment]
Backtraces sent to console on errors
OS Shell

**Comment by rabbibotton** [comment]
Support ANSI colors in terminal

![Screenshot 2024-04-12 074425](https://github.com/rabbibotton/clog/assets/11616789/97dd4159-c056-4b1d-aebe-e212e4af5ec5)

**Comment by rabbibotton** [comment]
New API for clog, clog:parent return the clog object that was used to create the current clog-obj and descendent of clog-element will have this set, others YMMV

**Comment by rabbibotton** [comment]
emacs style tabs using ctrl-t (mac)/alt-t or can be configured to use the Tab key

**Comment by rabbibotton** [comment]
made the emacs style tabs default now
Redid the repl so now have a area for working on your code and a button to transfer over to REPL, default now uses a per repl console that open with the repl

**Comment by rabbibotton** [comment]
Eval windows time out and close, the time is configurable for sel, form and file

**Comment by rabbibotton** [comment]
Better ECL compatibility

**Comment by rabbibotton** [comment]
Huge speedup for projects, projects now load "reasonably" well on windows

**Comment by rabbibotton** [comment]
Added Options -> Start SWANK Server Once
You can now startup a swank server on the builder (I no longer use emacs at all, but sometimes want to), let's you  connect on demand an emacs server (in theory lem also but the release versions are not working)

**Reply by tluyben** [reply]
You are using CLOG almost fully to develop CLOG? Are you pushing to get enough features in to never start up emacs? What is missing to get there?

**Reply by rabbibotton** [reply]
I fully use the CLOG Builder. I only fire up emacs for testing it when used with emacs. I am working now on features to take it beyond emacs :)

**Comment by rabbibotton** [comment]
Now Builder's main interface is the project tree. 
Project tree speed use on all platforms. Windows now very usable, asdf/quickslip load taking place in background and more progress hunting down the speed issues.

![project-tree](https://github.com/rabbibotton/clog/assets/11616789/46f77617-d63b-4668-9ebd-f006cf327998)

**Comment by rabbibotton** [comment]
clog-tool:open-file an alternative start for the builder and a command line script open-file for opening files

Right click menus for the project tree

**Comment by rabbibotton** [comment]
Replaced Dir View with Dir Tree that is similar to the project tree
